app/models/permitted_params.rb

```diff
@@ -303,8 +303,9 @@ def project(instance = nil)
   end
 
   def time_entry
-    params.fetch(:time_entry, {}).permit(:hours, :comments, :work_package_id,
-                                       :activity_id, :spent_on, custom_field_values: [])
+    params.fetch(:time_entry, {}).permit(
+      :hours, :comments, :work_package_id, :activity_id, :spent_on,
+      custom_field_values: custom_field_keys)
```

Comment:
Why are you not using what is already in place for that for other customized objects, e.g

https://github.com/opf/openproject/blob/fix/time_entry_permitted_params/app/models/permitted_params.rb#L173